extensions/amp-script/0.1/amp-script.js

```diff
@@ -131,7 +125,45 @@ export class AmpScript extends AMP.BaseElement {
       onReceiveMessage: data => {
         dev().info(TAG, 'From worker:', data);
       },
-    }).then(workerDom => {
+      onMutationPump: this.mutationPump_.bind(this),
+      onLongTask: promise => {
+        this.userActivation_.expandLongTask(promise);
+        // TODO(dvoytenko): consider additional "progress" UI.
+      },
+      sanitizer: {
+        sanitize: node => {
+          // DOMPurify sanitizes unsafe nodes by detaching them from parents.
+          // So, an unsafe `node` that has no parent will cause a runtime error.
+          // To avoid this, wrap `node` in a <div> if it has no parent.
+          const useWrapper = !node.parentNode;
```

Comment:
IMHO, this is a lot of code to inline...

Reply:
Good point. Encapsulated into a `SanitizerImpl` private class.